src/utils.ts

```diff
@@ -134,44 +156,47 @@ const pickByHashCode = (code: number, type: keyof DefaultOptions, opts?: PickByH
   return myDefaultOptions[index]
 }
 
-export const genConfig: GenConfigFunc = (userConfig = {}) => {
-  const isSeedConfig = typeof userConfig === 'string';
-  const hashCode = isSeedConfig && stringToHashCode(userConfig) || 0
-  const response = {} as Required<AvatarFullConfig>;
-  response.sex = isSeedConfig ? pickByHashCode(hashCode, 'sex') as Sex : (userConfig.sex || pickRandomFromList(defaultOptions.sex));
-  response.faceColor = isSeedConfig ? pickByHashCode(hashCode, 'faceColor') : (userConfig.faceColor || pickRandomFromList(defaultOptions.faceColor));
-  response.earSize = isSeedConfig ? pickByHashCode(hashCode, 'earSize') as EarSize : (userConfig.earSize || pickRandomFromList(defaultOptions.earSize));
-  response.eyeStyle = isSeedConfig ? pickByHashCode(hashCode, 'eyeStyle') as EyeStyle : (userConfig.eyeStyle || pickRandomFromList(defaultOptions.eyeStyle));
-  response.noseStyle = isSeedConfig ? pickByHashCode(hashCode, 'noseStyle') as NoseStyle : (userConfig.noseStyle || pickRandomFromList(defaultOptions.noseStyle));
-  response.mouthStyle = isSeedConfig ? pickByHashCode(hashCode, 'mouthStyle') as MouthStyle : (userConfig.mouthStyle || pickRandomFromList(defaultOptions.mouthStyle));
-  response.shirtStyle = isSeedConfig ? pickByHashCode(hashCode, 'shirtStyle') as ShirtStyle : (userConfig.shirtStyle || pickRandomFromList(defaultOptions.shirtStyle));
-  response.glassesStyle = isSeedConfig ? pickByHashCode(hashCode, 'glassesStyle', { usually: ["none"] }) as GlassesStyle : (userConfig.glassesStyle || pickRandomFromList(defaultOptions.glassesStyle, { usually: ["none"] }));
+export const genConfig: GenConfigFunc = (mixed?: AvatarFullConfig | string, seedConfig?: AvatarFullConfig): Required<AvatarFullConfig> => {
+  const isSeedConfig = typeof mixed === 'string'
+  const hashCode = isSeedConfig ? stringToHashCode(mixed) : 0
+  const response = (isSeedConfig ? {
+    ...createAvatarByHashCode(hashCode),
+    ...seedConfig
```

Comment:
Thanks for the PR, but I feel the seed config might not necessary and making things too complicated

One case I can imagine for the seed config is:
I have the user's email and sex, so I can specify the sex like `genConfig('my_email@test.com', { sex: 'man' })`, but here `isSeedConfig ? { ...createAvatarByHashCode(hashCode), ...seedConfig}` can't make it because we just simply merge the `sex`, in the final it might become a config with sex=man & hairStyle=womanLong

Anyway to be honst, I don't wanna adding seed config feature becuase I think it make seed too complicated.
I feel sorry about it because looks you spent a lot of time and efforts on this PR, maybe you can have this feature in your own branch or another lib

Reply:
Thank you for your reply! I like this library so much that I will add this PR to my own repo. Thank you again for your repo！